src/flags.rs

```diff
@@ -0,0 +1,478 @@
+            "-Cprofile-generate" => {
+                self.profile_generate = Some(value.expect("-Cprofile-generate must have a value"));
+            }
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#profile-use
+            "-Cprofile-use" => {
+                self.profile_use = Some(value.expect("-Cprofile-use must have a value"));
+            }
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#control-flow-guard
+            "-Ccontrol-flow-guard" => self.control_flow_guard = value.or(Some("true".into())),
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#lto
+            "-Clto" => self.lto = value.or(Some("true".into())),
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#relocation-model
+            "-Crelocation-model" => {
+                self.relocation_model = Some(value.expect("-Crelocation-model must have a value"))
+            }
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#embed-bitcode
+            "-Cembed-bitcode" => {
+                self.embed_bitcode = value.map_or(Some(true), |val| arg_to_bool(val.as_ref()));
+            }
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#force-frame-pointers
+            "-Cforce-frame-pointers" => {
+                self.force_frame_pointers =
+                    value.map_or(Some(true), |val| arg_to_bool(val.as_ref()));
+            }
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#link-dead-code
+            "-Clink-dead-code" => {
+                self.link_dead_code = value.map_or(Some(true), |val| arg_to_bool(val.as_ref()))
+            }
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#no-redzone
+            "-Cno-redzone" => {
+                self.no_redzone = value.map_or(Some(true), |val| arg_to_bool(val.as_ref()));
+            }
+            // https://doc.rust-lang.org/rustc/codegen-options/index.html#soft-float
+            // Note: This flag is now deprecated in rustc.
+            "-Csoft-float" => {
+                self.soft_float = value.map_or(Some(true), |val| arg_to_bool(val.as_ref()));
+            }
+            // https://doc.rust-lang.org/beta/unstable-book/compiler-flags/branch-protection.html
+            // FIXME: Drop the -Z variant and update the doc link once the option is stabilised
+            "-Zbranch-protection" | "-Cbranch-protection" => {
+                self.branch_protection =
+                    Some(value.expect("-Zbranch-protection must have a value"));
+            }
+            _ => {}
+        }
+    }
+
+    // Rust and clang/cc don't agree on what equivalent flags should look like.
+    pub fn cc_flags(
+        &self,
+        build: &Build,
+        path: &PathBuf,
+        family: &ToolFamily,
+        target: &Target,
+    ) -> Vec<OsString> {
```

Comment:
```suggestion
    pub fn cc_flags(
        &self,
        build: &Build,
        path: &PathBuf,
        family: ToolFamily,
        target: &Target,
        args: &mut Vec<OsString>,
    ) {
```

How about passing args directly so that we can push directly into it, and pass family as value since it's copyable and very small (much smaller than pointer)?

Alternatively, we can pass the `tool: &mut Tool` here, whichever is more convenient.

Reply:
Or just make push_if_supported into an inner function instead of a closure

Reply:
I'm ok with both honestly, one potential problem with removing the param, is that you can't push to the `args` directly and use the function at the same time.

If that doesn't happen, then yeah removing the param seems to be cleaner.